pyls/workspace.py

```diff
@@ -74,21 +74,24 @@ class Workspace(object):
     M_SHOW_MESSAGE = 'window/showMessage'
     PRELOADED_MODULES = get_preferred_submodules()
 
-    def __init__(self, root_uri, lang_server=None):
+    def __init__(self, root_uri, lang_server):
         self._root_uri = root_uri
         self._root_uri_scheme = uris.urlparse(self._root_uri)[0]
         self._root_path = uris.to_fs_path(self._root_uri)
         self._docs = {}
         self._lang_server = lang_server
 
         # Whilst incubating, keep private
-        self.__rope = Project(self._root_path)
+        if self._lang_server.config.plugin_settings('rope').get('create_folder', False):
```

Comment:
Get the user setting here does not work because the user settings are only loaded by the server after this function to be called.